go-controller/pkg/node/gateway.go

```diff
@@ -239,10 +239,27 @@ func gatewayInitInternal(nodeName, gwIntf, egressGatewayIntf string, gwNextHops
 			return nil, nil, err
 		}
 
+		ovsHardwareOffloadEnabled, err := util.IsOvsHwOffloadEnabled()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		reason := ""
 		if !chkPktLengthSupported {
-			klog.Warningf("OVS on this node does not support check packet length action in kernel datapath. This "+
-				"will cause incoming packets destined to OVN and larger than pod MTU: %d to the node, being dropped "+
-				"without sending fragmentation needed", config.Default.MTU)
+			reason += "OVS on this node does not support check packet length action in kernel datapath. "
+		}
+		/* This is a work around. In order to have the most optimal performance, the packet MTU check should be
```

Comment:
Could you add a `TODO` in the comment?
something like remove this workaround when check_pkt_length hardware offload is in place. It would be good if we can point to the WIP patches or issues.

Reply:
@zshi-redhat , Done! Added the documentation and TODO.